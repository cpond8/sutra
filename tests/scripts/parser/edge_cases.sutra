;; Parser edge cases test
;; Tests various syntax edge cases

;; Multiple expressions on same line
(print "first") (print "second")

;; Nested expressions
(print (+ 1 (+ 2 3)))

;; Mixed types in expressions
(print (list 1 "text" true))

;; Comments and whitespace
;; This is a comment
(print "after comment")

  ;; Indented comment
  (print "indented expression")

;; Empty lists
(print (list))

;; Deep nesting
(print (+ 1 (* 2 (+ 3 4))))
