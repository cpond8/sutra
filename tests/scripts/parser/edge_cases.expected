first
second
6
(1 text true)
after comment
indented expression
()
15
